data/mods/gen1/random-teams.ts

```diff
@@ -342,6 +348,112 @@ export class RandomGen1Teams extends RandomGen2Teams {
 			gender: false,
 		};
 	}
+
+	randomHCTeam(): PokemonSet[] {
+		const team = [];
+
+		const movePool = Object.keys(this.dex.data.Moves);
+		const typesPool = ['Bird', ...Object.keys(this.dex.data.TypeChart)];
+
+		const random6 = this.random6Pokemon();
+		const hackmonsCup: {[k: string]: HackmonsCupEntry} = {};
+
+		for (let i = 0; i < 6; i++) {
+			// Choose forme
+			const species = this.dex.getSpecies(random6[i]);
+			if (!hackmonsCup[species.id]) {
+				hackmonsCup[species.id] = {
+					types: [this.sample(typesPool), this.sample(typesPool)],
+					baseStats: {
+						hp: Utils.clampIntRange(this.random(256), 1),
+						atk: Utils.clampIntRange(this.random(256), 1),
+						def: Utils.clampIntRange(this.random(256), 1),
+						spa: Utils.clampIntRange(this.random(256), 1),
+						spd: 0,
+						spe: Utils.clampIntRange(this.random(256), 1),
+					},
+				};
+				hackmonsCup[species.id].baseStats.spd = hackmonsCup[species.id].baseStats.spa;
+			}
+			if (hackmonsCup[species.id].types[0] === hackmonsCup[species.id].types[1]) {
+				hackmonsCup[species.id].types.splice(1, 1);
+			}
+
+			// Random unique moves
+			const moves = [];
+			do {
+				const moveid = this.sampleNoReplace(movePool);
+				const move = this.dex.getMove(moveid);
+				if (move.gen <= this.gen && !move.isNonstandard && !move.name.startsWith('Hidden Power ')) {
+					moves.push(moveid);
+				}
+			} while (moves.length < 4);
+
+			// Random EVs
+			const evs = {hp: 0, atk: 0, def: 0, spa: 0, spd: 0, spe: 0};
+			const s: StatName[] = ['hp', 'atk', 'def', 'spa', 'spd', 'spe'];
+			for (const x of s) {
+				if (x === 'spd') continue;
+				evs[x] = this.random(256);
+			}
```

Comment:
```suggestion
			const evs = {
				hp: this.random(256),
				atk: this.random(256),
				def: this.random(256),
				spa: this.random(256),
				spd: 0,
				spe: this.random(256)
			};
```
Is there a reason why this wouldn't work? It's a lot cleaner.

Reply:
that's how current gen hackmons cup does it

Reply:
It looks like it does that because of some Gen 6 shenanigans (apparently Gen 6 Hackmons Cup has a 510-EV cap, but other gens remove that). Let's go with my suggestion, then; it does the same thing while being more readable and better performing.